test/TestProjects/MgmtSafeFlatten/Generated/Models/LayerOneBaseType.Serialization.cs

```diff
@@ -30,16 +31,7 @@ internal static LayerOneBaseType DeserializeLayerOneBaseType(JsonElement element
                     case "LayerOneFoo": return LayerOneFooType.DeserializeLayerOneFooType(element);
                 }
             }
-            LayerOneTypeName name = default;
-            foreach (var property in element.EnumerateObject())
-            {
-                if (property.NameEquals("name"))
-                {
-                    name = new LayerOneTypeName(property.Value.GetString());
-                    continue;
-                }
-            }
-            return new LayerOneBaseType(name);
+            throw new NotSupportedException("Deserialization of abstract type 'global::MgmtSafeFlatten.Models.LayerOneBaseType' not supported.");
```

Comment:
Would this be an issue when the service supports a new one for a customer using the old SDK? We should make sure there is a fallback class here

Reply:
Added the fallback internal class.